// asposecellscloud/dist/internal/auth.js
"use strict";
/*
* MIT License
*
* Copyright (c) 2025 Aspose.Cells Cloud
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*
*/
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth = void 0;
const requestHelper_1 = require("./requestHelper");
/**
 * Implements OAuth authentication
 */
class OAuth {
    /**
     * Apply authentication settings to header and query params.
     */
    applyToRequest(requestOptions, configuration) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.accessToken == null) {
                yield this._requestToken(configuration);
            }
            const now = new Date().getTime();
            if (now - this.getTime > 86300) {
                yield this._requestToken(configuration);
            }
            if (requestOptions && requestOptions.headers) {
                requestOptions.headers.Authorization = "Bearer " + this.accessToken;
            }
            return Promise.resolve();
        });
    }
    /**
     * Handle 401 response.
     */
    handle401response(configuration) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this._refreshToken(configuration);
        });
    }
    _requestToken(configuration) {
        return __awaiter(this, void 0, void 0, function* () {
            const requestOptions = {
                method: "POST",
                json: true,
                uri: configuration.baseUrl + (configuration.apiVersion == "v1.1" ? "/oauth2/token" : "/connect/token"),
                form: {
                    grant_type: "client_credentials",
                    client_id: configuration.clientId,
                    client_secret: configuration.clientSecret,
                },
            };
            const response = yield (0, requestHelper_1.invokeApiMethod)(requestOptions, configuration, true);
            this.accessToken = response.body.access_token;
            this.refreshToken = response.body.refresh_token;
            this.getTime = new Date().getTime();
            return Promise.resolve();
        });
    }
    _refreshToken(configuration) {
        return __awaiter(this, void 0, void 0, function* () {
            const requestOptions = {
                method: "POST",
                json: true,
                uri: configuration.baseUrl + "/connect/token",
                form: {
                    grant_type: "refresh_token",
                    refresh_token: this.refreshToken,
                },
            };
            const response = yield (0, requestHelper_1.invokeApiMethod)(requestOptions, configuration, true);
            this.accessToken = response.body.access_token;
            this.refreshToken = response.body.refresh_token;
            return Promise.resolve();
        });
    }
}
exports.OAuth = OAuth;
