// asposecellscloud/dist/internal/configuration.js
"use strict";
/*
* MIT License
*
* Copyright (c) 2025 Aspose.Cells Cloud
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
const auth_1 = require("../internal/auth");
const defaultBasePath = "https://api.aspose.cloud";
const defaultApiVersion = "v3.0";
/**
 * Words API configuration
 */
class Configuration {
    constructor(clientId, clientSecret, apiVersion, baseUrl, debugMode) {
        /**
         * Base Url.
         */
        this.baseUrl = defaultBasePath;
        /**
         * Base version.
         */
        this.apiVersion = defaultApiVersion;
        if (baseUrl) {
            this.baseUrl = baseUrl;
        }
        if (apiVersion) {
            this.apiVersion = apiVersion;
        }
        this.clientId = clientId;
        this.clientSecret = clientSecret;
        this.debugMode = debugMode;
        this.authentication = new auth_1.OAuth();
    }
    /**
     * Returns api base url
     */
    getApiBaseUrl() {
        return this.baseUrl + "/";
    }
}
exports.Configuration = Configuration;
